### DISCUSSION TITLE
Cannot destructure property 'publicRuntimeConfig'

### DISCUSSION THREAD

**Discussion by pangxiaolong** [vercel/next.js#24137] [Help]
### What version of Next.js are you using?

10.1.0

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

I can’t get the publicRuntimeConfig and serverRuntimeConfig in next.config.js through getConfig() in my index.js file. The initial version of my project is 8, and I have been upgrading to 9 in the future, and the version before 10.0.4 will not This problem occurs, I really have this problem in the process from 10.0.4 to 10.1.0

### Expected Behavior

I tried every version after 10.0.6 and the error appeared

### To Reproduce

The following is part of my code
index.ts
```
import next from 'next'
import getConfig from 'next/config'

const app: any = next({ dev })
//consoloe.log('getConfig',getConfig)  Function

const { publicRuntimeConfig, serverRuntimeConfig } = getConfig()
//consoloe.log('publicRuntimeConfig',publicRuntimeConfig)   undefind
```

next.config.js
```
const path = require('path')
const webpack = require('webpack')
const withPlugins = require('next-compose-plugins')

const configPath = path.join(
  __dirname,
  `./shared/env.${process.env.ENV ? (process.env.ENV === 'production' ? 'production' : 'testing') : 'development'}.js`
)

const publicRuntimeConfig = require(configPath)

module.exports = withPlugins(
  [
    ......
  ],
  {
    webpack(config, options) {
    ……
    publicRuntimeConfig,
    serverRuntimeConfig: {
      SESSION_COOKIE_SECRET: 'ZJrEPf……',
    }
  }
)
```

env.development.js
```

…

module.exports = {
  DAM_MAX_FONT_SIZE,
  API_TIMEOUT_INTERVAL: 60 * 1000,
  DOWNLOAD_TIMEOUT_INTERVAL: 60 * 1000,
  BASE_URL: 'http://……/pro',
  COOKIE_NAME: '……',
  COOKIE_DOMAIN: '.…….pub',
}
```

**Comment by ghost** [comment]
I am also experiencing the same

**Reply by ghost** [reply]
Hi @pangxiaolong, Did you find any solution?

**Reply by ghost** [reply]
Hi @timneutkens this is same as https://github.com/vercel/next.js/issues/25699, seeking your thoughts whether it is actually a bug ?
